### DISCUSSION TITLE
Is user_meta a good place to store if a users subscription is still active (stripe) and then use this store to make a policy to access premium content?

### DISCUSSION THREAD

**Discussion by ivan006** [supabase/supabase#15650] [Questions] [answered]
Is user_meta a good place to store if a users subscription is still active (stripe) and then use this store to make a policy to access premium content?

**Comment by GaryAustin1** [comment]
No.  user_metadata can be changed by the authenticated user at anytime.  You should store it in a secure table (similar to profile table) or in app_metadata which is secure.

**Reply by ivan006** [reply]
Thanks for the swift reply. But is the app_metadata  specific to the user? Alternatively can I relate a "profiles" tables to a user? If so can I use details of the related profile record to speak to my policy rules? If so how would i do that?

**Reply by GaryAustin1** [reply, marked as answer]
Yes the app meta data column is json and you can use it as long as you add to it and don't wipe what is there.  

The profiles table is normally using the user_id UUID as the primary key so is directly related to the user.  You can use RLS with auth.uid() on it.  If you don't let the user modify it then you can store your data there.  You use it in policies by doing join selects with auth.uid() on the profile table and check the column you are interested it.
This https://supabase.com/docs/guides/auth/row-level-security#policies-with-joins shows the basic idea behind using another table in RLS.

**Reply by ivan006** [reply]
Ok and i see i can make a table relate to an auth table then select user ok i didnt see that before

**Reply by ivan006** [reply]
I will try implement my requirements but in the mean time amazing work guys the system is jaw dropping
